comp110-worksheet-6/OxoBoard.cs

```diff
@@ -10,42 +10,125 @@ public enum Mark { None, O, X };
 
 	public class OxoBoard
 	{
-		// Constructor. Perform any necessary data initialisation here.
-		// Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
-		public OxoBoard(/* int width = 3, int height = 3, int inARow = 3 */)
+		Mark[,] boardSpaces = new Mark[3,3]; //Creates a new array using the Mark enumerable
+		public int fullSpaces; //creates an int value representing the filled spaces
+		List<Mark> occupiedSpacesX = new List<Mark>(); //Two seperate lists made to contain the used spaces for X and O
+		List<Mark> occupiedSpacesO = new List<Mark>();
+
+		public OxoBoard()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			boardSpaces.Initialize();
 		}
 
 		// Return the contents of the specified square.
 		public Mark GetSquare(int x, int y)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			if (Mark.X.Equals(true))
+			{
+				occupiedSpacesX.Add(GetSquare(x, y)); //Adds the currently occupied space of X to the occupiedSpacesX list
+				Mark test = new Mark(); //Sets test to represent Mark
+				return test; //Returns test
+			}
+
+			else
+			{
+				occupiedSpacesO.Add(GetSquare(x, y)); //Does the same for O to occupiedSpacesO
+				Mark test = new Mark();
+				return test;
+			}
+
+
 		}
 
 		// If the specified square is currently empty, fill it with mark and return true.
 		// If the square is not empty, leave it as-is and return False.
 		public bool SetSquare(int x, int y, Mark mark)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			if (Mark.None.Equals(true)) //If Mark.None is true, the space is empty, so the code will continue
+			{
+				boardSpaces[x, y] = mark; //fills the board spaces with mark
+				fullSpaces = fullSpaces + 1; //adds 1 to the fullSpaces value
+				return true;
+			}
+			else
+				return false;
 		}
 
 		// If there are still empty squares on the board, return false.
 		// If there are no empty squares, return true.
 		public bool IsBoardFull()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			if (fullSpaces == 9) //if the fullSpaces value is 9, all 9 spaces have been filled, so true is returned
+			{
+				return true;
+			}
+			else //if fullSpaces is not equal to 9 then not all spaces have been filled
+				return false;
+			
 		}
 
 		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
 		// Otherwise, return Mark.None.
 		public Mark GetWinner()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+			for (int length = 0; length < boardSpaces.GetLength(0); length++) //Sets 
```

Comment:
I see what you are trying to do here, and it's a reasonable idea -- just needs some more work in terms of getting the implementation right